qt/widgets/common/src/FindFilesWorker.cpp

```diff
@@ -122,6 +123,30 @@ void FindFilesWorker::run() {
   finishSearching(result);
 }
 
+/**
+ * Validate that the search text is not empty and contains no unsupported
+ * characters.
+ *
+ * @return a boolean indicating whether the search text is valid or not.
+ */
+bool FindFilesWorker::validateSearchText() {
```

Comment:
Can I ask how this manifests as a crash? It also feels like this should ask the `FileFinder` or `FileProperty` itself if the input is valid? If I imagine writing a different view then I would likely want to use the same model validation code.

Also, is it just ascii characters we want? Most modern linuxes will use `UTF-8` for file paths and Windows UTF-16. How exactly does the crash with chars like `&` manifest itself?

Reply:
A single full-stop '.' it gives a hard (sad mantid) crash on master - but interestingly I only saw this on the indirect data reduction interface (though I didn't try other indirect interfaces) but not on all the Load algorithms and the Eng Diff UI file browser worked - I assumed they use a different file-finder?

Reply:
The single full-stop only causes a crash in Indirect because the FileFinder returns that the '.' file is valid and has been found. The indirect data reduction interface checks to see if its valid using `filefinder->isValid()` and attempts to load the file immediately after enter is pressed, whereas I assume other interfaces will perform the load later on.

Reply:
Perhaps we want more than just ASCII characters? However when converting a QString from QLineEdit::text() to std::string it appeared that all characters other than ASCII characters where not being converted properly. i.e. QString(`£`) was converted to std::string(`Â£`), and this `Â` character caused the following error for me (in Debug mode) 

![image](https://user-images.githubusercontent.com/40830825/111473494-01477f00-8723-11eb-8098-716570d88d14.png)

Reply:
Ahh, maybe the above problem only happens because I am in debug mode?

Reply:
Ah yes you are correct that the framework only understand ascii but I'm curious as `std::string` is capable of holding the data do we know where this assertion is coming from?

Even though this pops up only in debug it's definitely a problem we need to solve. The Debug mode is basically saying something is wrong but Release mode will be doing something defined but most likey not what we want and gives these crashes.

Reply:
The assertion comes from the StringTokenizer on this line
https://github.com/mantidproject/mantid/blob/920de5ba6d08be16d933107ef57e6d4c5e189cef/Framework/Kernel/src/StringTokenizer.cpp#L17

Reply:
Ah okay. So I agree we need a check like this but what we're saying is that this is validation on what we are doing at the model level as we happen to be tokenizing it so I think the checking code live there and not in the widget.  What's the call stack before this?

Reply:
```
MantidKernel.dll!std::find_if_not<std::_String_iterator<std::_String_val<std::_Simple_types<char>>>,int (__cdecl*)(int)>(std::_String_iterator<std::_String_val<std::_Simple_types<char>>> _First, const std::_String_iterator<std::_String_val<std::_Simple_types<char>>> _Last, int(*)(int) _Pred) Line 419
	at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\algorithm(419)
MantidKernel.dll!`anonymous namespace'::trimTokenFromStart(std::string & s) Line 17
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(17)
MantidKernel.dll!`anonymous namespace'::trimToken(std::string & s) Line 28
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(28)
MantidKernel.dll!`anonymous-namespace'::splitIgnoringWhitespaceEmptyTokens::__l2::<lambda>(std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> first, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> second) Line 106
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(106)
MantidKernel.dll!`anonymous namespace'::for_each_token<std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>,std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>,void <lambda>(std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>)>(std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> first, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> last, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> s_first, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>> s_last, `anonymous-namespace'::splitIgnoringWhitespaceEmptyTokens::__l2::void <lambda>(std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>, std::_String_const_iterator<std::_String_val<std::_Simple_types<char>>>) binary_op) Line 52
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(52)
MantidKernel.dll!`anonymous namespace'::splitIgnoringWhitespaceEmptyTokens(const std::string & str, const std::string & delims) Line 110
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(110)
MantidKernel.dll!Mantid::Kernel::StringTokenizer::StringTokenizer(const std::string & str, const std::string & separators, unsigned int options) Line 149
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\Kernel\src\StringTokenizer.cpp(149)
MantidAPI.dll!Mantid::API::FileFinderImpl::findRuns(const std::string & hintstr, const std::vector<std::string,std::allocator<std::string>> & exts, const bool useExtsOnly) Line 561
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\Framework\API\src\FileFinder.cpp(561)
MantidQtWidgetsCommonQt5.dll!MantidQt::API::FindFilesWorker::run() Line 82
	at C:\Users\mlc47243\Documents\mantid-development\mantid-4\mantid\qt\widgets\common\src\FindFilesWorker.cpp(82)
Qt5Cored.dll!QThreadPoolThread::run() Line 101
	at c:\users\qt\work\qt\qtbase\src\corelib\thread\qthreadpool.cpp(101)
Qt5Cored.dll!QThreadPrivate::start(void * arg) Line 378
	at c:\users\qt\work\qt\qtbase\src\corelib\thread\qthread_win.cpp(378)
```

Reply:
I see, so the `isASCII` check should go somewhere in the StringTokenizer, and if the check fails I can throw an exception and catch it in the FindFilesWorker - which will then stop running and display the error?

Reply:
Nearly. I think `FileFinder` is the best place for adding some type of `isValidRunsInput` method that the widget can call the way you have now where it calls isascii directly. It would at this point just contain your ascii check but could also be used by other areas that want to check the validity of input into the findRuns function.

Reply:
Yes, the stack trace for the error should be in this thread @martyngigg